src/model/guild/system_channel.rs

```diff
@@ -1,14 +1,44 @@
-use bitflags::bitflags;
-use serde::{Deserialize, Serialize};
-
-bitflags! {
-    #[derive(Deserialize, Serialize)]
-    pub struct SystemChannelFlags: u32 {
-        /// To suppress the welcome messages.
-        const SUPPRESS_JOIN_NOTIFICATIONS = 1 << 0;
-        /// To suppress the server boosting messages.
-        const SUPPRESS_PREMIUM_SUBSCRIPTIONS = 1 << 1;
-        /// To supress the server tips.
-        const SUPPRESS_GUILD_REMINDER_NOTIFICATIONS = 1 << 2;
+use bitflags::__impl_bitflags;
+use serde::{Deserialize, Deserializer, Serialize, Serializer};
+
+use crate::internal::prelude::StdResult;
+use crate::model::utils::U64Visitor;
+
+/// Describes a system channel flags.
+#[derive(Copy, PartialEq, Eq, Clone, PartialOrd, Ord, Hash, Default)]
+#[cfg_attr(not(feature = "model"), derive(Debug, Deserialize, Serialize))]
```

Comment:
Why restrict the trait implementations based on the state of the `model` feature? The feature should only be applied on methods.
```suggestion
#[derive(Copy, PartialEq, Eq, Clone, PartialOrd, Ord, Hash, Default)]
```